packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -222,16 +218,13 @@ const withBlockBindingSupport = createHigherOrderComponent(
 			<>
 				{ Object.keys( bindings ).length > 0 && (
 					<BlockBindingBridge
-						blockProps={ props }
+						blockProps={ { ...props, attributes: boundAttributes } }
 						bindings={ bindings }
 						onPropValueChange={ updateBoundAttributes }
 					/>
 				) }
 
-				<BlockEdit
-					{ ...props }
-					attributes={ { ...props.attributes, ...boundAttributes } }
-				/>
+				<BlockEdit { ...props } attributes={ boundAttributes } />
```

Comment:
This is a dangerous change. By doing this, we would be harcoding the assignment of attributes for all blocks. It works only in the first renders because the reducer takes the block attributes value when initialized.

If, for example, you try to update the attributes of any block by dispatching the updateBlockAttributes action, it will fail. Once the block is rendered, the new changes in the store will not be reflected in the component.

Keep in mind that `boundAttributes` only should contain that: attributes that are bound to an external source. Probably, we should add a note.

Reply:
@retrofox 

So, guess that we'd need something like this:

```diff
diff --git a/packages/block-editor/src/hooks/use-bindings-attributes.js b/packages/block-editor/src/hooks/use-bindings-attributes.js
index b2bc7bdc9f..e5a757d674 100644
--- a/packages/block-editor/src/hooks/use-bindings-attributes.js
+++ b/packages/block-editor/src/hooks/use-bindings-attributes.js
@@ -204,6 +208,11 @@ const withBlockBindingSupport = createHigherOrderComponent(
 			props.attributes
 		);
 
+		// Update the boundAttributes variable when the block attributes change.
+		useEffect( () => {
+			updateBoundAttributes( props.attributes );
+		}, [ props.attributes ] );
+
 		/*
 		 * Create binding object filtering
 		 * only the attributes that can be bound.

```

But that would defeat the purpose of this PR 🙂 

Alternatively, we could technically use the same pattern of ["adjust the state directly during rendering" ](https://react.dev/learn/you-might-not-need-an-effect#:~:text=adjust%20the%20state%20directly%20during%20rendering) like:

```diff
diff --git a/packages/block-editor/src/hooks/use-bindings-attributes.js b/packages/block-editor/src/hooks/use-bindings-attributes.js
index b2bc7bdc9f..3732a799d9 100644
--- a/packages/block-editor/src/hooks/use-bindings-attributes.js
+++ b/packages/block-editor/src/hooks/use-bindings-attributes.js
@@ -204,6 +204,10 @@ const withBlockBindingSupport = createHigherOrderComponent(
 			props.attributes
 		);
 
+		if ( props.attributes !== boundAttributes ) {
+			updateBoundAttributes( props.attributes );
+		}
+
 		/*
 		 * Create binding object filtering
 		 * only the attributes that can be bound.
```

but I'm not 100% sure if this is safe either.

Reply:
As mentioned before, handling the data propagation relying on the component makes the approach fragile. It'll get tougher when we introduce the ability to edit the property value of the external source according to attribute changes.